src/flash-media-source.js

```diff
@@ -36,6 +37,12 @@ export default class FlashMediaSource extends videojs.EventTarget {
         }
       });
 
+      if (this.tech_.hls) {
+        this.tech_.hls.on('dispose', () => {
```

Comment:
I know that it looks like video.js' tech doesn't trigger dispose, but maybe we should consider adding it later, so we don't have to check for HLS specifically (or listening to a different type of event).